lib/bitvec.cpp

```diff
@@ -36,6 +36,7 @@ std::ostream &operator<<(std::ostream &os, const bitvec &bv) {
 }
 
 bitvec &bitvec::operator>>=(size_t count) {
+    assert(static_cast<int>(count) >= 0);
```

Comment:
This is wrong -- we can right shift by arbitrarily large amounts without a problem; if count is more than the largest bit (index) in the bitvec it will just result in a 0 (empty) bitvec and not any sort of problem.